src/index.tsx

```diff
@@ -356,6 +357,41 @@ class PyflyByWidget extends Widget {
           [PYFLYBY_START_MSG, joinedImports, PYFLYBY_END_MSG].join('\n')
         );
     }
+
+    this._cleanupNotebook();
+  }
+
+  _cleanupNotebook() {
+    const cells = this._context.model.cells;
+    const cellsToDelete: number[] = [];
+    for (let i = 0; i < cells.length; ++i) {
+      const cell = cells.get(i);
+      const lines = cell.sharedModel.getSource().split('\n');
+      let shouldDelete = true;
+      for (let j = 0; j < lines.length; ++j) {
+        const line = lines[j].trim();
+        // We are conservative here to only delete the cells which are empty or
+        // just have pyflyby messages
+        if (
+          line !== '' &&
+          line !== PYFLYBY_START_MSG &&
+          line !== PYFLYBY_END_MSG
+        ) {
+          shouldDelete = false;
+        }
+      }
+      if (shouldDelete) {
+        cellsToDelete.push(i);
+      }
+    }
+
+    this._context.model.sharedModel.transact(() => {
+      for (let i = 0; i < cellsToDelete.length; ++i) {
+        // we do cellsToDelete[i] - i because the cells will move back as we
+        // perform deletions
+        this._context.model.sharedModel.deleteCell(cellsToDelete[i] - i);
```

Comment:
This is hard to read. Better written as:
```
for(let i = cellsToDelete.length -1; i >=0; --i) {
        this._context.model.sharedModel.deleteCell(cellsToDelete[i]);
}
```

That is just go in reverse to avoid the issue

Reply:
Ah, makes sense